pkg/controller/machinepool/awsactuator.go

```diff
@@ -46,7 +49,62 @@ var (
 	noSuchSubnetRegex = regexp.MustCompile(`^InvalidSubnetID\.NotFound:\s+([^\t]+)\t`)
 )
 
-// NewAWSActuator is the constructor for building a AWSActuator
+func getSubnetIDByFilter(subnet *v1.AWSResourceReference, actuator AWSActuator) ([]*ec2.Subnet, error) {
+	req := &ec2.DescribeSubnetsInput{Filters: []*ec2.Filter{}}
+	for _, filter := range *subnet.Filters {
+		req.Filters = append(req.Filters,
+			&ec2.Filter{
+				Name:   aws.String(filter.Name),
+				Values: aws.StringSlice(filter.Values),
+			},
+		)
+	}
+	resp, err := actuator.awsClient.DescribeSubnets(req)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.Subnets) == 0 {
+		return nil, fmt.Errorf("no subnets found for filter %v", req.Filters)
+	}
+	// FIXME HIVE-2443 is it possible to have multiple subnets?
+	if len(resp.Subnets) > 1 {
+		return nil, fmt.Errorf("multiple subnets found for filter %v", req.Filters)
+	}
+
+	return resp.Subnets, nil
+}
+
+func (a *AWSActuator) GetRemoteMachineSetsWithOpFlags(pool *hivev1.MachinePool, remoteMachineSets *machineapi.MachineSetList, infrastructure *configv1.Infrastructure, logger log.FieldLogger) ([]msop.MachineSetWithOpFlags, error) {
+	remotes_to_update := []msop.MachineSetWithOpFlags{}
+	for _, rMS := range remoteMachineSets.Items {
+		fd1, err := FailureDomainFromProviderSpec(&rMS, infrastructure, logger)
+		if err != nil {
+			return nil, err
+		}
+
+		if aws.StringValue(fd1.AWS().Subnet.ID) == "" {
+			subnets, err := getSubnetIDByFilter(fd1.AWS().Subnet, *a)
+			if err != nil {
+				return nil, err
+			}
+			subnets, err = a.filterPublicSubnets(subnets, pool)
```

Comment:
This seems like it will break in validateSubnets, which makes sure we have exactly one public subnet per AZ in the pool, because here we're passing in just the one subnet we gleaned from the one remote MachineSet.